src/Microsoft.PowerShell.Commands.Utility/commands/utility/select-object.cs

```diff
@@ -419,18 +421,10 @@ private void ProcessParameter(MshParameter p, PSObject inputObject, List<PSNoteP
                 }
             }
 
+            // allow Select-Object -Property noexist-name to return a PSObject with property noexist-name,
```

Comment:
It would be clearer if you add quotes around `Select-Object -Property noexist-name`.